cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -251,266 +229,35 @@ While all the techniques referenced here do not require any user interaction, so
-
-    /**
-     * Application expected deployment domain: named "Target Origin" in OWASP CSRF article
-     */
-    private URL targetOrigin;
-
-    /***
-     * Secure generator
-     */
-    private final SecureRandom secureRandom = new SecureRandom();
-
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) 
-    throws IOException, ServletException {
-        HttpServletRequest httpReq = (HttpServletRequest) request;
-        HttpServletResponse httpResp = (HttpServletResponse) response;
-        String accessDeniedReason;
-
-        /* STEP 1: Verifying Same Origin with Standard Headers */
-        //Try to get the source from the "Origin" header
-        String source = httpReq.getHeader("Origin");
-        if (this.isBlank(source)) {
-            //If empty then fallback on "Referer" header
-            source = httpReq.getHeader("Referer");
-            //If this one is empty too then we trace the event and we block the request 
-            //(recommendation of the article)...
-            if (this.isBlank(source)) {
-                accessDeniedReason = "CSRFValidationFilter: ORIGIN and REFERER request" + 
-                "headers are both absent/empty so we block the request !";
-                LOG.warn(accessDeniedReason);
-                httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-                return;
-            }
-        }
+Most developers tend to ignore CSRF vulnerability on login forms as they assume that CSRF would not be applicable on login forms because user is not authenticated at that stage. That assumption is false. CSRF vulnerability can still occur on login forms where the user is not authenticated, but the impact/risk view for it is quite different from the impact/risk view of a general CSRF vulnerability (when a user is authenticated).
 
-        //Compare the source against the expected target origin
-        URL sourceURL = new URL(source);
-        if (!this.targetOrigin.getProtocol().equals(sourceURL.getProtocol()) || 
-            !this.targetOrigin.getHost().equals(sourceURL.getHost())
-        || this.targetOrigin.getPort() != sourceURL.getPort()) {
-            //One the part do not match so we trace the event and we block the request
-            accessDeniedReason = String.format("CSRFValidationFilter: Protocol/Host/Port " + 
-            "do not fully matches so we block the request! (%s != %s) ",
-                this.targetOrigin, sourceURL);
-            LOG.warn(accessDeniedReason);
-            httpResp.sendError(HttpServletResponse.SC_FORBIDDEN, accessDeniedReason);
-            return;
-        }
+With a CSRF vulnerability on login form, an attacker can make a victim login as them and learn behavior from their searches. For more information about login CSRF and other risks, see section 3 of [this](https://seclab.stanford.edu/websec/csrf/csrf.pdf) paper.
```

Comment:
A good example of this would be something like a user being logged in with CSRF, and then entering their credit card details to make a purchase (which are then stored in the attacker's account).